### PR TITLE
Schema compare Icon and other fixes

### PR DESCRIPTION
**Pull request by udeeshagautam**
This PR contains three fixes

1. Enable Icon for Schema Compare model view editor (both schema compare and SCMP files, although SCMP files are not parsed as SC page currently) - Here is how the new icons look

image

image

2. Changing to editable dropdown broke context setting i.e. if Schema compare is opened from a database it doesn't set source value automatically since Editable dropdown doesn't start with set value. Explicitly setting the dropdown display value.

3. Fixing a console error at component container item removal if description is null
`ERR Cannot read property 'id' of undefined: TypeError: Cannot read property 'id' of undefined
    at index.items.findIndex.item (file:///E:/Source/azuredatastudio/out/sql/workbench/electron-browser/modelComponents/componentBase.js:240:97)
    at Array.findIndex (<anonymous>)
    at FormContainer.removeFromContainer (file:///E:/Source/azuredatastudio/out/sql/workbench/electron-browser/modelComponents/componentBase.js:`

Targets line items in #5729

### CODE REVIEW

File: extensions/schema-compare/src/dialogs/schemaCompareDialog.ts [hunk 1]
```diff
@@ -421,7 +421,8 @@ export class SchemaCompareDialog {
 		this.sourceServerDropdown.onValueChanged(async (value) => {
 			if (this.sourceServerDropdown.values.findIndex(x => this.matchesValue(x, value)) === -1) {
 				this.sourceDatabaseDropdown.updateProperties({
-					values: []
+					values: [],
+					value: '  '
```
Comment (alanrenmsft) [inline extensions/schema-compare/src/dialogs/schemaCompareDialog.ts]: has to be whitespaces?

Comment (udeeshagautam) [reply extensions/schema-compare/src/dialogs/schemaCompareDialog.ts]: yeah... either real string or spaces... empty string or null is ignore and not updated

File: extensions/schema-compare/package.json [hunk 2]
```diff
@@ -34,6 +34,20 @@
 				}
 			}
 		],
+		"languages": [
+			{
+				"id": "scmp",
+				"filenames": [
+					"Schema Compare"
```
Comment (alanrenmsft) [inline extensions/schema-compare/package.json]: localize the strings in package.json

Comment (udeeshagautam) [reply extensions/schema-compare/package.json]: Localized everything other than file name. Thanks for the catch.

File: extensions/schema-compare/src/schemaCompareResult.ts [hunk 3]
```diff
@@ -53,7 +53,8 @@ export class SchemaCompareResult {
 		this.SchemaCompareActionMap[azdata.SchemaUpdateAction.Change] = localize('schemaCompare.changeAction', 'Change');
 		this.SchemaCompareActionMap[azdata.SchemaUpdateAction.Add] = localize('schemaCompare.addAction', 'Add');
 
-		this.editor = azdata.workspace.createModelViewEditor(localize('schemaCompare.Title', 'Schema Compare'), { retainContextWhenHidden: true, supportsSave: true });
+		// Do not localize paramater customResourceName
+		this.editor = azdata.workspace.createModelViewEditor(localize('schemaCompare.Title', 'Schema Compare'), { retainContextWhenHidden: true, supportsSave: true, customResourceName: 'Schema Compare' });
```
Comment (alanrenmsft) [inline extensions/schema-compare/src/schemaCompareResult.ts]: another thought, since you were able to set the title of the schema compare editor window's title to Schema Compare without needing this before, I guess the title and the resource name is not tied to each other, i am thinking you can give a more id/name like value here, e.g. _schema_compare. by doing this: 1. people won't think this value is localizable 2. it is unlikely that people will have a file named like this.  thoughts?

File: src/sql/azdata.proposed.d.ts [hunk 4]
```diff
@@ -3992,6 +3992,11 @@ declare module 'azdata' {
 		 * Does this model view editor support save?
 		 */
 		readonly supportsSave?: boolean;
+
+		/**
+		 * A custom name for this editor (To be used in icon setting etc.)
+		 */
+		readonly customResourceName?: string;
```
Comment (alanrenmsft) [inline src/sql/azdata.proposed.d.ts]: resourceName is good enough to me. also please be more descriptive here so that ppl can better understand how this works and why is this necessary

File: extensions/schema-compare/src/schemaCompareResult.ts (hunk 3 above)
Comment (udeeshagautam) [reply extensions/schema-compare/src/schemaCompareResult.ts]: good point.. will try with _schema_compare

File: src/sql/azdata.proposed.d.ts (hunk 4 above)
Comment (udeeshagautam) [reply src/sql/azdata.proposed.d.ts]: sure will explain the icon use case more... (I was not sure how else getResourceName() is used in general, that's why tried to be generic :) )

File: extensions/schema-compare/src/schemaCompareResult.ts (hunk 3 above)
Comment (udeeshagautam) [reply extensions/schema-compare/src/schemaCompareResult.ts]: as discussed leaving this one as is & documenting with comments
